scripts/test-sim-versions.go

```diff
@@ -0,0 +1,132 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+// +build ignore
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+)
+
+var SkippedUplinkVersions = []string{
+	"v0.10.1",
+	"v0.10.0", "v0.10.2",
+	"v0.11.0", "v0.11.1", "v0.11.2", "v0.11.3", "v0.11.4", "v0.11.5", "v0.11.6", "v0.11.7",
+	"v0.12.0", "v0.12.1", "v0.12.2", "v0.12.3", "v0.12.4", "v0.12.5", "v0.12.6",
+	"v0.13.0", "v0.13.1", "v0.13.2", "v0.13.3", "v0.13.4", "v0.13.5", "v0.13.6",
+}
+
+type VersionsTest struct {
+	Stage1 *Stage `yaml:"stage1"`
+	Stage2 *Stage `yaml:"stage2"`
+}
+
+type Stage struct {
+	SatelliteVersion    string   `yaml:"sat_version"`
+	UplinkVersions      []string `yaml:"uplink_versions"`
+	StoragenodeVersions []string `yaml:"storagenode_versions"`
+}
+
+func main() {
+	if err := run(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func run() error {
+	if len(os.Args) < 3 {
+		return errors.New("Please provide path to script file and yaml file via command line")
+	}
+
+	scriptFile := os.Args[1]
+	yamlFile := os.Args[2]
```

Comment:
Why do we need a yaml file instead of adding it directly into the code?

Reply:
I am not sure if I remember the reason (@isaachess might), but I think it could have been that a yaml file would be easier for someone like Jens to maintain.
We actually originally had the versions hardcoded into `test-sim-versions.sh`, then we wanted to split them out into a yaml file since maintaining them in bash was unsustainable. But importing a yaml file from bash is not trivial, so we created the go file to facilitate that. At that point, maybe we could have just forgotten about the yaml file entirely and done that stuff from Go. I don't have a strong preference one way or the other.

Reply:
maybe dwelling unnecessarily on this but is having the versions in the Go file more sustainable than having them in the bash file? I also don't have a strong opinion on where they go though

Reply:
Yingrong actually has modified the script to get the uplink, storagenode, and satellite so we don't need a config.yaml anymore. Basically we are planning on doing the following:
```
`latest_release` - the last actual release (not including the current/ongoing release)
`master` - the latest code (which the current/ongoing release will be based on)
`uplink_versions` - a list of uplink releases to test (e.g. `26.3, 25.3, 24.5, ... 15.4`)

`stage 1`
* set up a clean storj-sim based on the `latest_release`
* for every uplink in `uplink_versions`, upload three files (`bucket-v0.XX.X/file-#` )

`stage 2`
* start satellite on `master`
* start storagenodes 0-4 on `latest_release`
* start storagenodes 5-9 on `master`
* for every uplink in `uplink_versions`, download every file that was uploaded in `stage 1` and verify correctness
```

All the versions discussed above should be able to be acquired without having to maintain an actual set of versions somewhere.